src/main/java/com/hrznstudio/galacticraft/server/command/GalacticraftCommands.java

```diff
@@ -23,119 +23,251 @@
+
+            commandDispatcher.register(
+                    LiteralArgumentBuilder.<ServerCommandSource>literal("gcrlistbodies")
+                    .executes(GalacticraftCommands::listBodies));
+        });
+    }
+
+    private static int teleportToEarth(CommandContext<ServerCommandSource> context) {
+        final int[] retval = new int[1];
+        retval[0] = Command.SINGLE_SUCCESS;
+        // Clear the expired timers
+        for (UUID id : GCR_HOUSTON_TIMERS.keySet()) {
+            if (GCR_HOUSTON_TIMERS.get(id) + GCR_HOUSTON_TIMER_LENGTH < System.currentTimeMillis()) {
+                GCR_HOUSTON_TIMERS.remove(id);
+            }
+        }
+        context.getSource().getMinecraftServer().execute(() -> {
+            try {
+                ServerPlayerEntity player = context.getSource().getPlayer();
+                ServerWorld serverWorld = context.getSource().getMinecraftServer().getWorld(RegistryKey.of(Registry.DIMENSION, DimensionType.OVERWORLD_ID));
+                if (serverWorld == null) {
+                    context.getSource().sendError(new TranslatableText("commands.galacticraft-rewoven.dimensiontp.failure.dimension").setStyle(Style.EMPTY.withColor(Formatting.RED)));
+                    retval[0] = -1;
+                    return;
+                } else if (context.getSource().getWorld().equals(serverWorld)) {
+                    context.getSource().sendError(new TranslatableText("commands.galacticraft-rewoven.gcrhouston.on_earth_already").setStyle(Style.EMPTY.withColor(Formatting.RED)));
+                    retval[0] = -1;
+                    return;
+                }
+                UUID playerID = context.getSource().getPlayer().getGameProfile().getId();
+                if (!GCR_HOUSTON_TIMERS.containsKey(playerID)) {
+                    GCR_HOUSTON_TIMERS.put(playerID, System.currentTimeMillis());
+                    context.getSource().sendFeedback(new TranslatableText("commands.galacticraft-rewoven.gcrhouston.confirm", serverWorld.getRegistryKey().getValue()).setStyle(Style.EMPTY.withColor(Formatting.RED)), false);
+                } else if (GCR_HOUSTON_TIMERS.get(playerID) + GCR_HOUSTON_TIMER_LENGTH > System.currentTimeMillis()) {
+                    GCR_HOUSTON_TIMERS.remove(playerID);
+                    BlockPos pos = getValidTeleportPos(serverWorld, player);
+                    player.teleport(serverWorld,
+                            pos.getX(),
+                            pos.getY(),
+                            pos.getZ(),
+                            player.yaw,
+                            player.pitch);
+                    context.getSource().sendFeedback(new TranslatableText("commands.galacticraft-rewoven.gcrhouston.success", serverWorld.getRegistryKey().getValue()).setStyle(Style.EMPTY.withColor(Formatting.GREEN)), true);
+                }
+            } catch (CommandSyntaxException ignore) {
```

Comment:
Instead of doing a `try {} catch` maybe test if the player is specified?

Reply:
I'm not sure how I would do that.
All the other commands were wrapped in try/catches as well, it's probably better that way instead of crashing the game (or if it doesn't crash, the player won't get any feedback on why the command failed at least).
I'm pretty sure that the only place that error could possibly happen is in the first two lines, but I don't want to take that chance.
If you know exactly how to change it to make it better, I'll leave that to you :)

Reply:
Nevermind, made it more specific.